File: backend/src/main/java/net/pengcook/user/dto/ReportResponse.java
package net.pengcook.user.dto;

import java.time.LocalDateTime;
import net.pengcook.user.domain.Reason;
import net.pengcook.user.domain.Type;
import net.pengcook.user.domain.UserReport;

public record ReportResponse(
        long reportId,
        long reporterId,
        long reporteeId,
        Reason reason,
        Type type,
        long targetId,
        String details,
        LocalDateTime createdAt
) {
    public ReportResponse(UserReport userReport) {
        this(
                userReport.getId(),
                userReport.getReporter().getId(),
                userReport.getReportee().getId(),
                userReport.getReason(),
                userReport.getType(),
                userReport.getTargetId(),
                userReport.getDetails(),
                userReport.getCreatedAt()
        );
    }
}

File: android/app/src/main/java/net/pengcook/android/data/model/usercontrol/ReportResponse.kt
package net.pengcook.android.data.model.usercontrol

data class ReportResponse(
    val reportId: Long,
    val reporterId: Long,
    val reporteeId: Long,
    val reason: String,
    val type: String,
    val targetId: Long,
    val details: String?,
    val createdAt: String,
)
